app/server/api/company/[id].get.ts

```diff
@@ -0,0 +1,13 @@
+import { defineEventHandler } from 'h3'
+import type { CompatibilityEvent } from 'h3'
+import { parseDataAs, parseParamsAs, z } from '@sidestream-tech/nuxt-sidebase-parse'
+import { Company, companyFull } from '~/server/database/entities/Company'
+
+const paramsSchema = z.object({
+  id: z.string().regex(/^\d+$/).transform(Number),
+})
+
+export default defineEventHandler(async (event: CompatibilityEvent) => {
+  const params = parseParamsAs(event, paramsSchema)
+  return parseDataAs(Company.findOneOrThrow({ where: { id: params.id } }), companyFull)
```

Comment:
there's two ways to improve this:

a)
```suggestion
  const where = parseParamsAs(event, paramsSchema)
  return parseDataAs(Company.findOneOrThrow({ where }), companyFull)
```

b)
```suggestion
  const params = parseParamsAs(event, paramsSchema)
  return parseDataAs(Company.findOneOrThrow({ where: params }), companyFull)
```